packages/cyberstorm/src/components/MultiSelectSearch/MultiSelectSearch.tsx

```diff
@@ -0,0 +1,191 @@
+ */
+export const MultiSelectSearch = React.forwardRef<HTMLInputElement, Props>(
+  function MultiSelectSearch(props, ref) {
+    const { options, value, onChange, onBlur, placeholder, color } = props;
+    const menuRef = React.useRef<HTMLDivElement | null>(null);
+    const inputRef = React.useRef<HTMLInputElement | null>(null);
+
+    const [isVisible, setIsVisible] = React.useState(false);
+    const [search, setSearch] = React.useState("");
+    const [filteredOptions, setFilteredOptions] = React.useState(options);
+
+    function add(incomingOption: MultiSelectSearchOption) {
+      if (!value.some((option) => option.value === incomingOption.value)) {
+        onChange([...value, incomingOption]);
+      }
+    }
+
+    function remove(incomingOption: MultiSelectSearchOption) {
+      onChange(value.filter((option) => option.value !== incomingOption.value));
+    }
+
+    const hideMenu = React.useCallback(
+      (e: MouseEvent | TouchEvent) => {
+        if (
+          menuRef.current &&
+          isVisible &&
+          !menuRef.current.contains(isNode(e.target) ? e.target : null)
+        ) {
+          setIsVisible(false);
+          onBlur();
+        }
+      },
+      [setIsVisible, isVisible, onBlur, menuRef]
+    );
+
+    // Event listeners for closing menu when clicking or touching outside.
+    React.useEffect(() => {
+      document.addEventListener("mousedown", hideMenu);
+      document.addEventListener("touchstart", hideMenu);
+      return () => {
+        document.removeEventListener("mousedown", hideMenu);
+        document.removeEventListener("touchstart", hideMenu);
+      };
+    });
+
+    React.useEffect(() => {
+      const updatedOptions = options.filter(
+        (option) =>
+          !value.includes(option) &&
+          option.label.toLowerCase().includes(search.toLowerCase())
+      );
+      setFilteredOptions(updatedOptions);
+    }, [value, search, options, setFilteredOptions]);
+
+    return (
+      <div className={styles.root} ref={ref}>
+        <div className={styles.selected}>
+          {value.map((option) => {
+            return (
+              <Button.Root
+                key={option.value}
+                onClick={() => remove(option)}
+                paddingSize="small"
+                style={{ gap: "0.5rem" }}
+              >
+                <Button.ButtonLabel>{option.label}</Button.ButtonLabel>
+                <Button.ButtonIcon>
+                  <FontAwesomeIcon icon={faXmark} />
+                </Button.ButtonIcon>
+              </Button.Root>
+            );
+          })}
+        </div>
+        <div
+          className={styles.search}
+          onFocus={(e) => {
+            inputRef.current && inputRef.current.focus();
+            e.stopPropagation();
+          }}
+          role="button"
+          tabIndex={0}
+          ref={menuRef}
+        >
+          <div className={styles.inputContainer} data-color={color}>
+            <input
```

Comment:
On Storybook, at least with Firefox, the tab navigation will now get stuck into this input, not moving forward no matter how many times I tab.

Reply:
Fixed this issue, but I'll keep the keyboard navigation todo there, because I'm not 100% what is the best practice, so we need to fully fix the keyboard nav later.